Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/AvoidNoWarnAnalyzerSuppressionAnalyzer.cs

```diff
@@ -0,0 +1,163 @@
+// © 2025 Koninklijke Philips N.V. See License.md in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Philips.CodeAnalysis.Common;
+
+namespace Philips.CodeAnalysis.MaintainabilityAnalyzers.Maintainability
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class AvoidNoWarnAnalyzerSuppressionAnalyzer : SolutionAnalyzer
+	{
+		private const string Title = @"Avoid NoWarn for analyzer suppression";
+		private const string MessageFormat = @"Use .editorconfig instead of NoWarn project setting to suppress analyzer '{0}'. Configure with 'dotnet_diagnostic.{0}.severity = none'.";
+		private const string Description = @"NoWarn project settings should be avoided for analyzer suppression. Use .editorconfig files for better maintainability and team consistency.";
+
+		private static readonly char[] SeparatorChars = { ';', ',' };
+
+		public AvoidNoWarnAnalyzerSuppressionAnalyzer()
+			: base(DiagnosticId.AvoidNoWarnAnalyzerSuppression, Title, MessageFormat, Description, Categories.Maintainability, isEnabled: false)
+		{
+		}
+
+		public override void Initialize(AnalysisContext context)
+		{
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+			context.EnableConcurrentExecution();
+
+			context.RegisterCompilationAction(AnalyzeCompilation);
+		}
+
+		private void AnalyzeCompilation(CompilationAnalysisContext context)
+		{
+			try
+			{
+				var projectFilePath = TryFindProjectFileFromSourcePaths(context);
+				if (string.IsNullOrEmpty(projectFilePath) || !File.Exists(projectFilePath))
+				{
+					return;
+				}
+
+				AnalyzeProjectFile(context, projectFilePath);
+			}
+			catch (Exception)
```

Comment:
Remember dogfood builds, which will complain about an empty catch block. In this scenario, unless you know of a reason this could happen, I'm tempted to advise trying the removal of it, and seeing how it performs. (We don't generally wrap analyzers in blank try...catch blocks (but maybe we should??)